test/service/test_request_validation.py

```diff
@@ -165,23 +165,43 @@ def test_file_order(self):
     @mock_sts
     @mock_s3
     def test_manifest(self):
+        self._index_canned_bundle(("f79257a7-dfc6-46d6-ae00-ba4b25313c10", "2018-09-14T133314.453337Z"))
         # moto will mock the requests.get call so we can't hit localhost; add_passthru let's us hit the server
         # see this GitHub issue and comment: https://github.com/spulec/moto/issues/1026#issuecomment-380054270
         with ResponsesHelper() as helper:
             helper.add_passthru(self.base_url)
             storage_service = StorageService()
             storage_service.create_bucket()
 
-            url = self.base_url + '/repository/files/export?filters={"file":{}}'
-            response = requests.get(url)
-            self.assertEqual(200, response.status_code, 'Unable to download manifest')
-            tsv_file = csv.DictReader(response.iter_lines(decode_unicode=True), delimiter='\t')
-            # 2 because self.bundle has 2 files
-            self.assertEqual(len(list(tsv_file)), 2, 'Wrong number of files were found.')
-            manifest_config = json.load(open('{}/request_config.json'.format(self.service_config_dir), 'r'))['manifest']
-            expected_fieldnames = list(manifest_config['bundles'].keys()) + list(
-                manifest_config['contents.files'].keys())
-            self.assertEqual(expected_fieldnames, tsv_file.fieldnames, 'Manifest headers are not configured correctly')
+            for single_part in "0", "1":
+                with self.subTest(is_single_part=single_part):
+                    with mock.patch.dict(os.environ, AZUL_DISABLE_MULTIPART_MANIFESTS=single_part):
+                        url = self.base_url + '/repository/files/export?filters={"file":{}}'
+                        response = requests.get(url)
+                        self.assertEqual(200, response.status_code, 'Unable to download manifest')
+                        tsv_file = csv.DictReader(response.iter_lines(decode_unicode=True), delimiter='\t')
+                        rows = {r['file_uuid']: r for r in tsv_file}
+                        # 7 because self.bundle and bundle f79257a7-dfc6-46d6-ae00-ba4b25313c10 have 7 files total.
+                        self.assertEqual(len(rows.items()), 7, 'Wrong number of files were found.')
+                        # Fieldnames from the bundles entity are always the first headers in the manifest. The rest of
+                        # the fieldnames are ordered as they are mapped in request_config.json from top to bottom.
+                        expected_fieldnames = ["bundle_uuid", "bundle_version", "file_content_type", "file_name",
```

Comment:
Checkout the bdbag test and how it asserts the first column literally. You could also assert the entire tsv literall in one giant assertEquals(). freeze() is your friend.

Reply:
I tried copying that test last week, but the test manifest generated isn't the same between tests. If a manifest with the exact same files is generated, its cells containing concatenated values (such as '123 || 345') are not written the same order. Should I sort these values before concatenating them?

Reply:
I think that would be a good idea.